### DISCUSSION TITLE
Added students to a closed semester

### DISCUSSION THREAD

**Discussion by scottcraig** [bytedeck/bytedeck#1567] [General]
I added courses to some students manually, as they were in more than one block. The form asked to select the semester.

Unfortunately, I had previously closed a semester by accident, so I created a new one. The new one was given the same name (feb-2024) and I can't see a way to distinguish them. It would be nice to see an id or some other way to distinguish them. A student listing from each semester would be useful here, too.

Looking at the semesters page from the Admin menu, I see that there are now 5 students in the closed semester. It would have been better to not have been able to select the closed semester to add a studentcourse to.

I think I can work around this by going through each student and editing their courses. The drop-down in that page only has one choice for semester, and I assume it it the open one.

**Comment by scottcraig** [comment]
I was able to recover by going through the list of students and editing all students with more than one course. I edited each of the courses for these students by choosing the only semester that appeared in the dropdown.

This also fixed an issue with grades for students with more than one course. Previously, (when a student has a course from the open semester and another from the closed semester of the same name) the XP was not being dived by 2, but showing the full amount for each course. It seems to be working correctly now.

**Comment by scottcraig** [comment]
A minor related issue: The semester page under the Admin tab has a table with a column labelled "# Students". The number shown is actually the number of StudentCourses, so it overcounts the students, as some of them are taking more than one course.

**Comment by tylerecouture** [comment]
@scottcraig thanks for posting here.  The Semester naming is definitely not helpful.  You should be able to change the start date of the closed semester to another month, and that will give it a different name so you can tell them apart.  You can do that via Admin > Django Admin then find the Semesters table, where you can edit stuff on the item. (Since you won't be able to edit from the main site once closed).  Just be careful in Django Admin because it can mess everything up if you delete the wrong thing, nothing is protected back there!

**Comment by tylerecouture** [comment]
@scottcraig Also, you are correct, the Active semester is the only option on the course registration form.

**Comment by tylerecouture** [comment]
@scottcraig "It would have been better to not have been able to select the closed semester to add a studentcourse to."  Can you clarify what you mean here?  Only the active semester should be available when you are adding a student to a course:
![image](https://github.com/bytedeck/bytedeck/assets/10604391/e333aae7-4e20-4f5f-b085-cc217ce902fc)


If you are talking about "Django Admin".  There be dragons and you are on your own =)  I recommend not using it unless you have to, it's a place to go as a last resort.
